src/librustc/diagnostics.rs

```diff
@@ -1923,14 +1932,14 @@ From [RFC 246]:
 "##,
 
 E0395: r##"
-The value assigned to a constant expression must be known at compile time,
+The value assigned to a constant scalar must be known at compile time,
 which is not the case when comparing raw pointers. Erroneous code example:
 
 ```
 static foo: i32 = 42;
 static bar: i32 = 43;
 
-static baz: bool = { (&foo as *const i32) == (&bar as *const i32) };
+static baz: bool = { (&foo as *const i32) < (&bar as *const i32) };
```

Comment:
I prefer not make the issue more clear rather than leaving the subtle points of linking.